docs/sysext.md

```diff
@@ -0,0 +1,129 @@
+# Building system extensions with mkosi
+
+[System extension](https://uapi-group.org/specifications/specs/extension_image/)
+images may – dynamically at runtime — extend the base system with an
+overlay containing additional files.
+
+To build system extensions with mkosi, we first have to create a base
+image on top of which we can build our extension.
+
+To keep things manageable, we'll use mkosi's support for building
+multiple images so that we can build our base image and system extension
+in one go.
+
+Let's start by defining a base configuration file `mkosi.conf` with some
+shared settings:
+
+```conf
+[Output]
+OutputDirectory=mkosi.output
+CacheDirectory=mkosi.cache
+```
+
+Now let's continue with the base image definition by writing the
+following to `mkosi.images/base/mkosi.conf`:
+
+```conf
+[Output]
+Format=directory
+
+[Content]
+CleanPackageMetadata=no
+Packages=systemd
+         udev
+```
+
+We use the `directory` output format here instead of the `disk` output
+so that we can build our extension without needing root privileges.
+
+Now that we have our base image, we can define a sysext that builds on
+top of it by writing the following to `mkosi.images/btrfs/mkosi.conf`:
+
+```conf
+[Config]
+Dependencies=base
+
+[Output]
+Format=disk
+Overlay=yes
+
+[Content]
+BaseTrees=%O/base
+Packages=btrfs-progs
+```
+
+`BaseTrees=` point to our base image and `Overlay=yes` instructs mkosi
+to only package the files added on top of the base tree.
+
+We'll also want to mark our extension as a system extension. To do that,
+write the following to
+`mkosi.images/btrfs/mkosi.extra/usr/lib/extension-release.d/extension-release.btrfs`:
+
+```conf
+ID=<distribution>
+VERSION_ID=<distribution-version>
+ARCHITECTURE=<architecture>
+SYSEXT_SCOPE=initrd
```

Comment:
This is a bit specialised and it makes sense for the example of btrfs, but a short paragraph on this at the beginning would probably be helpful for casual readers, what the example here is, otherwise I think this might be cargo culted.

Reply:
I think any sysext would need at least ID, VERSION_ID and SYSEXT_SCOPE=initrd if it has to be an initrd extension. ARCHITECTURE can be dropped, and maybe SYSEXT_SCOPE=initrd should be documented (or at least a link to a doc should be there) to show the reader the options available there